File: android/app/src/main/java/com/alamada/childmonitor/models/Notification.java
package com.alamada.childmonitor.models;

import com.google.gson.annotations.SerializedName;

public class Notification {
    @SerializedName("id")      public int     id;
    @SerializedName("title")   public String  title;
    @SerializedName("message") public String  message;
    @SerializedName("sent_at") public String  sentAt;
    @SerializedName("is_read") public boolean isRead;
}

File: backend/app/Models/Notification.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Notification extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'message',
        'recipient_group',
        'recipient_id',
        'sent_by',
        'sent_at',
    ];

    protected $casts = [
        'sent_at' => 'datetime',
    ];

    public function sender()
    {
        return $this->belongsTo(User::class, 'sent_by');
    }

    public function recipient()
    {
        return $this->belongsTo(User::class, 'recipient_id');
    }

    public function readers()
    {
        return $this->belongsToMany(User::class, 'notification_reads', 'notification_id', 'user_id')
                    ->withPivot('read_at')
                    ->withTimestamps();
    }
}
